Helm Charts for DEV Version of OpenCloud

ferenc-hechler opened the pull request:
The Helm chart is mainly following the getting-started description for Docker:
https://docs.opencloud.eu/docs/admin/getting-started/docker/docker/

In the installation instructions is an example use-case with port-forwarding from localhost:
https://github.com/ferenc-hechler/opencloud-helm?tab=readme-ov-file#-installing-the-helm-charts

michaelstingl commented:
## Tested OpenCloud Helm Charts with Rancher Desktop on macOS

I've successfully tested the Helm charts in this branch with Rancher Desktop on macOS. The minimal approach works perfectly for local testing and Kubernetes deployment development.

### My Setup
- macOS Sequoia 15.4
- Rancher Desktop 1.18.2 with Kubernetes 1.32.3
- Container Engine: moby
- OpenCloud version 2.0.0

### Steps Performed
1. Installed Rancher Desktop
2. Enabled Kubernetes in Rancher Desktop
3. Created namespace: `kubectl create namespace opencloud`
4. Installed OpenCloud: `helm install opencloud -n opencloud ./charts/opencloud`
5. Verified pod status: `kubectl get pods -n opencloud -w`
6. Set up port forwarding: `kubectl port-forward -n opencloud svc/opencloud-service 9200:443`
7. Accessed OpenCloud at https://localhost:9200

### Observations
- The pod started up correctly
- PVCs were created and bound successfully
- Web interface is accessible and login works with admin/admin
- Logs show normal operation (only expected TLS handshake errors)
- Rancher Desktop's built-in dashboard provides good visibility into all resources

### Suggested README Improvement
I'd suggest making it more explicit in the README that these charts are primarily intended for local Kubernetes deployment development and testing environments, not for production deployments. Perhaps adding a note at the top like:

```
> **Note:** These charts are primarily intended for Kubernetes deployment development and testing environments, not for production use. They provide a simplified setup with minimal configuration.
```

This would help set clear expectations for users and prevent misuse in production environments without proper security, resource management, and high availability considerations.

I'd be happy to contribute detailed documentation for setting up Rancher Desktop on macOS as a development environment for OpenCloud Kubernetes deployments.

ferenc-hechler commented:
Thanks @michaelstingl for your feedback.
As there is now an elaborated PR for a full fledged helm chart I renamed my proposal helm chart to `opencloud-dev`
to make clear, that it is thought for quick development purposes only.

I added you as collaborator to my fork.
Happy if you can add hints for Rancher Desktop on MacOS deployments.   :-)

michaelstingl commented:
@ferenc-hechler Thanks for adding me as a collaborator to your fork! 

I've now contributed the detailed documentation for Rancher Desktop on macOS as discussed. I created a separate file `RANCHER-DESKTOP-MACOS.md` in the charts/opencloud-dev directory that provides comprehensive guidance for macOS users.

The documentation includes:
- Comparison of Kubernetes options for macOS (Rancher Desktop, Docker Desktop, minikube, k3d)
- Step-by-step setup instructions for Rancher Desktop
- Detailed installation and configuration steps
- Monitoring and debugging tips
- Troubleshooting section
- Reference comparison table

You can find my PR here: https://github.com/ferenc-hechler/opencloud-helm/pull/1

This should help macOS users get started quickly with the OpenCloud DEV chart using Rancher Desktop.

michaelstingl commented:
While testing again for my PR, I found two YAML formatting issues that prevented installation with Rancher Desktop on macOS:

1. **Issue in `Chart.yaml`**: The multi-line description wasn't properly formatted for YAML, resulting in a error when converting YAML to JSON.
   ```
   Error: INSTALLATION FAILED: cannot load Chart.yaml: error converting YAML to JSON: yaml: line 6: mapping values are not allowed in this context
   ```

2. **Issue in `opencloud-data-pvc.yaml`**: Indentation under `annotations` used tabs instead of spaces, causing a YAML parsing error.
   ```
   Error: INSTALLATION FAILED: YAML parse error on opencloud-dev/templates/opencloud-data-pvc.yaml: error converting YAML to JSON: yaml: line 6: found character that cannot start any token
   ```

**Solution:**
- Formatted the description in `Chart.yaml` as a single-line string with quotes
- Changed indentation in the PVC file from tabs to spaces

I've pushed these fixes directly to @ferenc-hechler's repository so the chart can now be installed correctly.
